File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/RemoteControl.java
package com.example.try_gameengine.remotecontroller;

import android.view.MotionEvent;
import com.example.try_gameengine.remotecontroller.RemoteController.CommandType;

/**
 * @author irons
 *
 */
public class RemoteControl {
	Command slot;
	Command[] onCommands;
	Command[] offCommands;

	public RemoteControl() {
		onCommands = new Command[7];
		offCommands = new Command[7];
		Command noCommand = new NoCommand();
		for (int i = 0; i < 7; i++) {
			onCommands[i] = noCommand;
			offCommands[i] = noCommand;
		}
	}

	public void setCommand(Command command) {
		slot = command;
	}

	public void setCommand(int slot, Command onCommand, Command offCommand) {
		onCommands[slot] = onCommand;
		offCommands[slot] = offCommand;
	}

	public CommandType executePressDown(float x, float y,
			int motionEventPointerId, MotionEvent event) {
		CommandType commandType = CommandType.None;
		for (Command command : onCommands) {
			if (command.checkExecute(x, y, event)) {
				commandType = command.execute();
				command.setMotionEventPointerId(motionEventPointerId);
				break;
			}
		}
		return commandType;
	}

	public CommandType executePressUp(float x, float y,
			int motionEventPointerId, MotionEvent event) {
		CommandType commandType = CommandType.None;

		for (Command command : offCommands) {
			if (command.checkExecute(x, y, event)) {
				commandType = command.execute();
				break;
			}
		}

		return commandType;
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/RemoteControl.kt
package com.example.try_gameengine.remotecontroller

import android.view.MotionEvent

/**
 * @author irons
 // */
class RemoteControl {
    var slot: Command? = null
    var onCommands: Array<Command>
    var offCommands: Array<Command>

    init {
        val noCommand: Command = NoCommand()
        onCommands = Array(7) { noCommand }
        offCommands = Array(7) { noCommand }
    }

    fun setCommand(command: Command?) {
        slot = command
    }

    fun setCommand(slot: Int, onCommand: Command?, offCommand: Command?) {
        onCommands[slot] = onCommand!!
        offCommands[slot] = offCommand!!
    }

    fun executePressDown(
        x: Float, y: Float,
        motionEventPointerId: Int, event: MotionEvent?
    ): RemoteController.CommandType? {
        var commandType: RemoteController.CommandType? = RemoteController.CommandType.None
        for (command in onCommands) {
            if (command.checkExecute(x, y, event)) {
                commandType = command.execute()
                command.setMotionEventPointerId(motionEventPointerId)
                break
            }
        }
        return commandType
    }

    fun executePressUp(
        x: Float, y: Float,
        motionEventPointerId: Int, event: MotionEvent?
    ): RemoteController.CommandType? {
        var commandType: RemoteController.CommandType? = RemoteController.CommandType.None

        for (command in offCommands) {
            if (command.checkExecute(x, y, event)) {
                commandType = command.execute()
                break
            }
        }

        return commandType
    }
}
